src/snowflake/snowpark/_internal/analyzer/analyzer_utils.py

```diff
@@ -585,8 +585,11 @@ def create_table_statement(
     )
 
 
-def insert_into_statement(table_name: str, child: str) -> str:
-    return f"{INSERT}{INTO}{table_name} {project_statement([], child)}"
+def insert_into_statement(
+    table_name: str, column_names: Optional[Iterable[str]], child: str
```

Comment:
nit: shall we give a default value None?
```suggestion
    table_name: str, child: str, column_names: Optional[Iterable[str]] = None
```